### DISCUSSION TITLE
Hello, I have a question about router handle response.

### DISCUSSION THREAD

**Discussion by sanghogu** [vercel/next.js#55001] [App Router] [answered]
Hello, this is my first time creating an app with node. I want to develop it as standard as possible.

When handling a router, is it okay to use call new Response instead of the NextResponse object?
When I try it, it works fine.

use reason: for using new Response is to redirect relative paths and return custom body.

**Comment by devjiwonchoi** [comment, marked as answer]
Yes. It is absolutely fine to use the `Response` object.
Route handlers should be fully compatible with it.

> Route Handlers allow you to create custom request handlers for a given route using the Web [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) and [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) APIs.

One of the examples using `Response`:

```ts
import { cookies } from 'next/headers'
 
export async function GET(request: Request) {
  const cookieStore = cookies()
  const token = cookieStore.get('token')
 
  return new Response('Hello, Next.js!', {
    status: 200,
    headers: { 'Set-Cookie': `token=${token.value}` },
  })
}
```
